lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -1,25 +1,67 @@
-require "pry"
+require "active_support/inflector"
 
 module RuboCop
   module Cop
     module RootCops
       class FactoryFileName < Cop
-        FILE_NAME_ERROR = "Factory name should match file name".freeze
+        # Whitelisted words feel like invariant nouns but they are really not
+        # Ex: the singular of data is datum, but we should be able to name a factory '*_data' without error
+        WHITELIST = ["data".freeze, "metadata".freeze].freeze
 
         def investigate(processed_source)
           file_path = processed_source.buffer.name
           @base_file_name = File.basename(file_path, ".rb")
+          @base_file_name_body = _generate_body(@base_file_name)
+          @base_file_name_last_word = @base_file_name.split("_").last
         end
 
         def on_send(node)
-          method_name = node.method_name
+          return if _file_name_has_error?
+
+          receiver, method_name = *node
+          _, receiver_name = *receiver
+
+          if receiver_name == :FactoryBot && method_name == :define
+            unless _file_name_pluralized?
```

Comment:
We can put this within line 24's condition to save some nesting: `if receiver_name == :FactoryBot && method_name == :define && !_file_name_pluralized?`